peps/pep-0736.rst

```diff
@@ -0,0 +1,361 @@
+keyword-only function definitions. This is less arbitrary than ``f(%x)`` or
+variants, but no less so than ``f(x=)``. 
+
+However, we object that:
+
+* For any given argument, it is less clear from local context whether it is
+  positional or named. The ``*`` could easily be missed in a long argument list
+  and named arguments may be read as positional or vice versa.
+* It is unclear whether keyword arguments for which the value was not elided may
+  follow the ``*``. If so, then their relative position will be inconsistent but
+  if not, then an arbitrary grouping is enforced between different types of
+  keyword arguments.
+
+Objections
+==========
+
+There are only a few hard objections to the introduction of this syntactic
+sugar. Most of those not in favour of this feature are in the camp of 'I
+wouldn't use it'. However, over the extensive conversations about this feature,
+the following objections were the most common:
+
+The syntax is ugly
+------------------
+
+This objection is by far the most common. On the contrary, we argue that:
+
+* This objection is is subjective and many community members disagree.
+* A nearly-identical syntax is already established for f-strings.
+* Programmers will, as ever, adjust over time.
+
+The feature is confusing
+------------------------
+
+We argue that:
+
+* Introducing new features typically has this impact temporarily.
+* The syntax is very similar to the established ``f'{x=}'`` syntax.
+* The feature and syntax are familiar from other popular modern languages.
+* The expansion of ``x=`` to ``x=x`` is in fact a trivial feature and inherently
+  significantly less complex than ``*arg`` and ``**kwarg`` expansion.
+* This particular syntactic form has been independently proposed on numerous
+  occasions, indicating that it is the most obvious [8]_ [9]_ [13]_.
+
+The feature is not explicit
+---------------------------
+
+This is based on a misunderstanding of the Zen of Python. Keyword arguments are
+fundamentally more explicit than positional ones where argument assignment is
+only visible at the function definition. On the contrary, the proposed syntactic
+sugar contains all the information as is conveyed by the established keyword
+argument syntax but without the redundancy. Moreover, the introduction of this
+syntactic sugar incentivises use of keyword arguments, making typical Python
+codebases more explicit.
+
+The feature adds another way of doing things
+--------------------------------------------
+
+The same argument can be made against all syntax changes. This is a simple
+syntactic sugar, much as ``x += 1`` is sugar for ``x = x + 1`` when ``x`` is an
+integer. This isn't tantamount to a 'new way' of passing arguments but a more
+readable notation for the same way.
+
+Renaming the variable in the calling context will break the code
+----------------------------------------------------------------
+
+A ``NameError`` would make the mistake abundantly clear. Moreover, text editors
```

Comment:
How should editors behave if, while renaming the variable, the new name already exists in the scope where the function is being employed?